### DISCUSSION TITLE
Does fluent nhibernate support async?

### DISCUSSION THREAD

**Discussion by jt4000** [nhibernate/fluent-nhibernate#454] [General] [answered]
Hello, 
Does fluent nhibernate support async?

Thank you!

**Comment by i4004** [comment]
Hello! To execute database queries you use `NHibernate` part, not `FluentNhibernate` side, and yes, it supports async, for example:

```csharp
var queryable = session.Query<MyObject>();

var obj = queryable.FirstOrDefaultAsync();
```

session - `ISession` for `NHIbernate`.

More examples: https://github.com/SimplifyNet/Simplify/blob/master/src/Simplify.FluentNHibernate/SessionExtensions.cs

**Comment by hazzik** [comment, marked as answer]
Fluent NHibernate is a mapping tool and query side is handled by NHibernate. NHibernate does support async.
